docs/howto.md

```diff
@@ -0,0 +1,292 @@
+may not be everyone's first choice. We're looking into other methods in the
+future. Do note that by separating the credentials in the secret, the password
+is never directly accessed by the operator itself.
+
+
+# Create shares that are accessible outside the cluster
+
+Unless you took extra steps on your own, the shares created in the previous
+examples are only accessible by other processes running within the same
+Kubernetes cluster.  The Samba Operator does support a simple method for
+automatically exposing the shares outside of the cluster, this involves
+creating an SmbCommonConfig resource.  Create an SmbCommonConfig like the one
+below - that specifies the value "external" for the `publish:` key under
+`network:`. Then create an SmbShare that refers to this in it's `commonConfig:`
+field. When the operator processes this SmbShare, it will also create a
+Kubernetes Service configured for load balancing.
+
+```yaml
+apiVersion: samba-operator.samba.org/v1alpha1
+kind: SmbCommonConfig
+metadata:
+  name: mypublished
+spec:
+  network:
+    publish: external
+```
+```yaml
+apiVersion: samba-operator.samba.org/v1alpha1
+kind: SmbShare
+metadata:
+  name: myshare
+spec:
+  commonConfig: mypublished
+  readOnly: false
+  storage:
+    pvc:
+      spec:
+        accessModes:
+          - ReadWriteOnce
+        resources:
+          requests:
+            storage: 1Gi
+```
+
+When the [LoadBalancer
+Service](https://kubernetes.io/docs/concepts/services-networking/service/#loadbalancer)
+is created it will report the IP/hostname that you can use to access the share
+when you run `kubectl get services`.
+
+
+# Create shares accessible outside the cluster with DNS registration
+
+This example is like the previous but includes automatically registering the
+external IP address used to access the share in the Active Directory DNS. This
+means that hosts connected to the AD DNS do not need to be told the IP of the
+LoadBalancer Service. This is only supported on shares enabled for Active
+Directory.
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: join1
+type: Opaque
+stringData:
+  # Change the value below to match the username and password for a user that
+  # can join systems your test AD Domain
+  join.json: |
+    {"username": "Administrator", "password": "P4ssw0rd"}
+```
+```yaml
+apiVersion: samba-operator.samba.org/v1alpha1
+kind: SmbSecurityConfig
+metadata:
+  name: mydomain
+spec:
+  mode: active-directory
+  realm: cooldomain.myorg.example.com
+  joinSources:
+  - userJoin:
+      secret: join1
+      key: join.json
+  dns:
+    register: external-ip
+```
+```yaml
+apiVersion: samba-operator.samba.org/v1alpha1
+kind: SmbCommonConfig
+metadata:
+  name: mypublished
+spec:
+  network:
+    publish: external
+```
+```yaml
+apiVersion: samba-operator.samba.org/v1alpha1
+kind: SmbShare
+metadata:
+  name: myshare
+spec:
+  securityConfig: mydomain
+  readOnly: false
```

Comment:
@phlogistonjohn commonconfig needs to be included here

Reply:
Fixed. Thanks for catching that!